pgml-cms/docs/api/sql-extension/pgml.rank.md

```diff
@@ -0,0 +1,34 @@
+---
+description: Rank documents against a piece of text using the specified ranking model.
+---
+
+# pgml.rank()
+
+The `pgml.rank` function is used to rank text documents against some text. This function is primarly used as the last step in a search system where the results returned from the initial search are reranked before being used.
+
+## API
+
+```postgresql
+pgml.rank(
+    transformer TEXT,  -- transformer name
+    query TEXT,        -- text to rank against
+    documents TEXT[],  -- documents to rank
+    kwargs JSON        -- optional arguments (see below)
+)
+```
+
+## Example
+
+```postgresql
+SELECT pgml.rank('mixedbread-ai/mxbai-rerank-base-v1', 'test', ARRAY['doc1', 'doc2']);
+```
+
+```postgresql
+SELECT pgml.chunk('mixedbread-ai/mxbai-rerank-base-v1', 'test', ARRAY['doc1', 'doc2'], '{"return_documents": false, "top_k": 10}'::JSONB);
+```
+
+## Supported Ranking Models
+
```

Comment:
Let's provide a bit more information if available. To have good documentation, we need to be a bit more descriptive about how, and why things work.

Reply:
How does this look?https://github.com/postgresml/postgresml/pull/1514/commits/f9c5ff2e82ecd96ffef0daaa08c92bb57099032c

Reply:
Me like